typescript/packages/jumble/integration/utils.ts

```diff
@@ -8,9 +11,56 @@ export const decode = (() => {
   return (buffer: Uint8Array): string => decoder.decode(buffer);
 })();
 
+const RECORD_SNAPSHOTS = false;
+const SNAPSHOTS_DIR = join(Deno.cwd(), "test_snapshots");
+console.log("SNAPSHOTS_DIR=", SNAPSHOTS_DIR);
+
+// Helper function to assert, take screenshot and snapshot HTML
+export async function assertAndSnapshot(
+  condition: unknown,
+  message: string,
+  page?: Page | void,
+  snapshotName?: string,
+): Promise<void> {
+  if (!condition) {
+    throw new Error(message);
+  }
+  assert(condition, message);
+
+  if (RECORD_SNAPSHOTS && page && snapshotName) {
+    ensureDirSync(SNAPSHOTS_DIR);
+    const timestamp = new Date().toISOString().replace(/[:.]/g, "-");
+    const filePrefix = `${snapshotName}_${timestamp}`;
+
+    // Take screenshot
+    const screenshot = await page.screenshot();
+    Deno.writeFileSync(`${SNAPSHOTS_DIR}/${filePrefix}.png`, screenshot);
+
+    // Snapshot HTML
+    const html = await page.content();
+    Deno.writeTextFileSync(`${SNAPSHOTS_DIR}/${filePrefix}.html`, html);
+
+    console.log(`Snapshot saved: ${filePrefix}`);
+  }
+}
+
 export const sleep = (ms: number) =>
   new Promise((resolve) => setTimeout(resolve, ms));
 
+export async function tryClick(
+  el?: ElementHandle | null,
+  page?: Page,
+): Promise<void> {
+  await assertAndSnapshot(
+    el !== null && el !== undefined,
+    "Element does not exist or is not clickable",
+    page,
+    "try_click_element",
+  );
+
+  await el!.click();
```

Comment:
Does the assertion eliminate the need for the `!`?

Reply:
Now that I am using the `asserts` keyword, yes!